app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1002,12 +1003,14 @@ export default Controller.extend(bufferedProperty("model"), {
     },
 
     changeNotice(post) {
+      const modal = this.modal;
       return new Promise(function (resolve, reject) {
-        const modal = showModal("change-post-notice", { model: post });
-        modal.setProperties({
-          resolve,
-          reject,
-          notice: post.notice ? post.notice.raw : "",
+        modal.show(ChangePostNoticeModal, {
```

Comment:
As far as I can tell, this promise wrapper only exists so that the [widgets will rerender](https://github.com/discourse/discourse/blob/f5e8e737ad3fd7939f2c102d9dcbc5948294e0bf/app/assets/javascripts/discourse/app/widgets/widget.js#L353-L359) once the operation is complete and the modal is closed. The new Modal API returns a promise, which is automatically resolved whenever the modal is closed. So I think we could simplify this entire `changeNotice` action down to

```js
async changeNotice(post){
  await this.modal.show(ChangePostNoticeModal, { model: { post } });
}
```
And then remove all the custom resolve/reject logic from the component.

Does that sound feasible?

Reply:
Yes. I confirmed during development that the promise is exactly for re-rendering the widget. I'll try this sugared version and see. If it works it's certainly a lot better. 🙂

Reply:
This does indeed work. 🙇